modules/questions.py

```diff
@@ -121,139 +153,147 @@ def is_review_request(self, message: ServiceMessage) -> Optional[SetQuestionQuer
+            return SetQuestionByAtOrReplyQuery(
+                self.review_msg_id2question_ids[cast(str, msg_ref_id)],
                 "Live on site",
             )
 
     ####################
     # Post question(s) #
     ####################
 
-    async def message_post_question(
+    async def cb_post_question(
         self, query: PostOrCountQuestionQuery, message: ServiceMessage
     ) -> Response:
         """Post message to Discord for least recently asked question.
         It will contain question title and GDoc url.
         """
         # get questions df
-        questions = self.get_questions_df(query)
-
-        # if tags were specified, filter questions on those which have at least one of the tags
+        questions_df = coda_api.questions_df
+
+        # if status was specified, filter questions for that status
+        if query.status:
+            questions_df = questions_df.query("status == @query.status")
+        else: # otherwise, filter for question that ain't Live on site
+            questions_df = questions_df.query("status != 'Live on site'")
+        # if tag was specified, filter for questions having that tag
         if query.tag:
-            questions = query.filter_on_tag(questions)
+            questions_df = query.filter_on_tag(questions_df)
 
         # get all the oldest ones and shuffle them
-        questions = shuffle_questions(questions)
-        questions = get_least_recently_asked_on_discord(questions)
+        questions_df = get_least_recently_asked_on_discord(questions_df)
+        questions_df = shuffle_questions(questions_df)
 
-        # get exactly n questions (default is 1)
-        questions = query.filter_on_max_num_of_questions(questions)
+        # get specified number of questions (default [if unspecified] is 1)
+        questions_df = query.filter_on_max_num_of_questions(questions_df)
 
         # make question message and return response
-        msg = query.next_result_info(len(questions))
-        if not questions.empty:
-            msg += "\n\n" + "\n---\n".join(
-                make_post_question_message(cast(ParsedRow, r.to_dict()))
-                for _, r in questions.iterrows()
+        response_text = query.post_result_info(len(questions_df))
+        if not questions_df.empty:
+            response_text += "\n\n" + "\n---\n".join(
+                make_post_question_message(cast(QuestionRow, r.to_dict()))
+                for _, r in questions_df.iterrows()
             )
 
-        # if there is only one question, remember that one
-        if len(questions) == 1:
-            self.last_question_id = questions.iloc[0]["id"]
+        # if there is only one question, cache its ID
+        if len(questions_df) == 1:
+            self.last_question_id = questions_df.iloc[0]["id"]
+
+        # in this particular case, I think it's better to send all the questions in advance, instead of one-by-one
+        await message.channel.send(response_text)
```

Comment:
- what is difference of using this, instead of having `text=response_test` in the returned Response object?
- would this introduce risk that two different modules both respond to query? (suppose another module gave confidence 9 to the query, somehow)

Reply:
If Stampy is asked to post too many questions, it takes some time to update all of them and this introduces a significant delay before invoking the module and posting it to the channel.
On the other hand, we probably don't want to spam the channel with 10 questions even if somebody asks for it, so it's probably better if I cap the number of questions to e.g. 5.